tecton_client/_internal/utils.py

```diff
@@ -66,3 +67,9 @@ def validate_request_args(
     if not feature_service_name:
         msg = "feature_service_name must be provided"
         raise ValueError(msg)
+
+
+def validate_url(url: str):
+    parse_result = urlparse(url)
+    if parse_result.scheme not in ("http", "https"):
+        raise ValueError("Invalid URL: Must begin with http:// or https://")
```

Comment:
I'm not sure I really understand the value of this check. Doesn't our HTTP client also validate this already:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_api.py", line 198, in get
    return request(
           ^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_api.py", line 106, in request
    return client.request(
           ^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_client.py", line 827, in request
    return self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_client.py", line 914, in send
    response = self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_client.py", line 942, in _send_handling_auth
    response = self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_client.py", line 979, in _send_handling_redirects
    response = self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_client.py", line 1015, in _send_single_request
    response = transport.handle_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_transports/default.py", line 232, in handle_request
    with map_httpcore_exceptions():
  File "/Users/brian/.local/share/pyenv/versions/3.11.6/lib/python3.11/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Users/brian/.local/share/brian-venvs/anyscale/lib/python3.11/site-packages/httpx/_transports/default.py", line 86, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.UnsupportedProtocol: Request URL is missing an 'http://' or 'https://' protocol.
```

Reply:
... I think I'll actually remove this though; the user isn't going to get any validation that the url _exists_ until they make an api call. Having different url validation happen at different times is also confusing.